org.eclipse.scanning.event/src/org/eclipse/scanning/event/ConsumerImpl.java

```diff
@@ -488,121 +475,107 @@ private void init() throws EventException {
 		// NOTE: Not all consumers check the submit queue and
 		// pause before they start.
-        checkStartPaused();
+		checkStartPaused();
 
 		// It is possible to call start() and then awaitStart().
 		if (latchStart!=null) latchStart.countDown();
 	}
 
-	private boolean consume() throws Exception {
+	private void consume() throws Exception {
 
 		checkPaused(); // blocks until not paused.
-		if (!isActive()) return false; // Might have pasued for a long time.
+		if (!isActive()) return; // Might have pasued for a long time.
 
 		// Consumes messages from the queue.
-	Message m = getMessage(uri, getSubmitQueueName());
-        if (m!=null) {
-		waitTime = 0; // We got a message
+		Message m = getMessage(uri, getSubmitQueueName());
+		if (m != null) {
+			waitTime = 0; // We got a message
 
-		// TODO FIXME Check if we have the max number of processes
-		// exceeded and wait until we don't...
-
-		TextMessage t = (TextMessage)m;
-
-		final String json  = t.getText();
-
-			@SuppressWarnings("unchecked")
-			final U bean   = (U) service.unmarshal(json, getBeanClass());
-
-		executeBean(bean);
-        }
-        return true;
+			// TODO FIXME Check if we have the max number of processes
+			// exceeded and wait until we don't...
+			TextMessage t = (TextMessage) m;
+			final String json = t.getText();
+			final U bean = service.unmarshal(json, getBeanClass());
+			executeBean(bean);
+		}
 	}
 
-	private boolean processException(Throwable ne) throws EventException {
+	private boolean processException(Throwable e) throws EventException {
+		LOGGER.debug("Processing error in consumer", e);
 
-		if (ne instanceof EventException || ne instanceof InterruptedException) {
-			if (Thread.interrupted()) return false;
-			logger.trace("Cannot consume message ", ne);
-			if (isDurable()) {
-				return true;
-			}
-			return false;
-		}
-
-		logger.debug("Error in consumer!", ne);
-		if (ne.getClass().getSimpleName().contains("Json")) {
-			logger.error("Fatal except deserializing object!", ne);
-			return true;
-		}
-		if (ne.getClass().getSimpleName().endsWith("UnrecognizedPropertyException")) {
-			logger.error("Cannot deserialize bean!", ne);
+		// if error occurred deserializing the bean, log a specific message and don't fail even if not durable
+		if (e.getClass().getSimpleName().contains("Json") || e.getClass().getSimpleName().endsWith("UnrecognizedPropertyException")) {
+			LOGGER.error("Could not deserialize bean.", e);
 			return true;
 		}
 
-		if (ne.getClass().getSimpleName().endsWith("ClassCastException")) {
-			logger.error("Problem with serialization?", ne);
+		// normal error case, log error message and continue if
+		if (e instanceof EventException || e instanceof InterruptedException) {
+			if (Thread.interrupted()) return false;
+			LOGGER.error("Cannot consume message ", e);
+			return isDurable();
 		}
 
-
 		if (!isDurable()) return false;
 
+		// TODO: below assumes some conenction problem with activemq. why would any remaining
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)